chickadee/processes/wps_bccaq.py

```diff
@@ -0,0 +1,147 @@
+    Full statistical downscaling of coarse scale global climate model (GCM)
+    output to a fine spatial resolution"""
+
+    def __init__(self):
+        self.status_percentage_steps = {
+            "start": 0,
+            "process": 10,
+            "build_output": 95,
+            "complete": 100,
+        }
+
+        inputs = [
+            LiteralInput(
+                "gcm_file",
+                "GCM NetCDF file",
+                abstract="GCM simulations in NetCDF format",
+                data_type="string",
+            ),
+            LiteralInput(
+                "obs_file",
+                "Observations NetCDF file",
+                abstract="high-res gridded historical observations",
+                data_type="string",
+            ),
+            LiteralInput(
+                "var",
+                "Variable to Downscale",
+                abstract="Name of the NetCDF variable to downscale",
+                data_type="string",
+            ),
+            LiteralInput(
+                "end_date",
+                "End Date",
+                abstract="Defines the end of the calibration period",
+                default="2005-12-31",
+                data_type="string",
+            ),
+            LiteralInput(
+                "out_file",
+                "Output File Name",
+                abstract="Path to output file",
+                data_type="string",
+            ),
+            log_level,
+        ]
+
+        outputs = [
+            ComplexOutput(
+                "output",
+                "Output",
+                abstract="output netCDF file",
+                supported_formats=[FORMATS.NETCDF],
+            ),
+        ]
+
+        super(BCCAQ, self).__init__(
+            self._handler,
+            identifier="bccaq",
+            title="BCCAQ",
+            abstract="Full statistical downscaling of coarse scale global climate model (GCM) output to a fine spatial resolution",
+            keywords=["downscaling"],
+            metadata=[
+                Metadata("PyWPS", "https://pywps.org/"),
+                Metadata("Birdhouse", "http://bird-house.github.io/"),
+                Metadata("PyWPS Demo", "https://pywps-demo.readthedocs.io/en/latest/"),
+            ],
+            version="0.1.0",
+            inputs=inputs,
+            outputs=outputs,
+            store_supported=True,
+            status_supported=True,
+        )
+
+    def _handler(self, request, response):
+        loglevel = request.inputs["loglevel"][0].data
+        log_handler(
+            self,
+            response,
+            "Starting Process",
+            logger,
+            log_level=loglevel,
+            process_step="start",
+        )
+
+        gcm_file = request.inputs["gcm_file"][0].data
+        obs_file = request.inputs["obs_file"][0].data
+        var = request.inputs["var"][0].data
+        end_date = request.inputs["end_date"][0].data
+        out_file = request.inputs["out_file"][0].data
```

Comment:
I expect this will change as we look to accept `thredds` inputs. But, in the future we could consider making this a loop since we are essentially doing the same this over and over again.